test/core.jl

```diff
@@ -4609,3 +4609,27 @@ end
 
 # issue #19963
 @test_nowarn ccall(:jl_free, Void, (Ptr{Void}, ), C_NULL)
+
+# Wrong string size on 64bits for large string.
+if Sys.WORD_SIZE == 64
+    @noinline function test_large_string20360(slot)
+        try
+            slot[] = Base._string_n(2^32)
```

Comment:
The hope is that it can always work, which is why a much lower level function is used to allocate the string. FWIW, this is smaller than the default heap size we use.
The allocation should success unless a tight VM limit is set, in which case it should raise a OOM error and pass the test.
Without touching the string, we shouldn't trigger the OOM killer after the allocation succeeded and the OS should usually happily hand back a large number of COW zero pages with only the first page dirty due to metadata stored in there. The `tester` function makes sure that the allocation is free'd immediately afterwards.